src/client/Client.js

```diff
@@ -492,6 +571,46 @@ class Client extends BaseClient {
     if (typeof options.retryLimit !== 'number' || isNaN(options.retryLimit)) {
       throw new TypeError('CLIENT_INVALID_OPTION', 'retryLimit', 'a number');
     }
+    if (options.cacheData !== undefined) {
+      if (typeof options.cacheData !== 'object') {
+        throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData', 'an object or undefined');
+      }
+      if (typeof options.cacheData.pollInterval !== 'number' || isNaN(options.cacheData.pollInterval)) {
+        throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.pollInterval', 'a number');
+      }
+      if (typeof options.cacheData.dataLifetime === 'number') {
+        if (isNaN(options.cacheData.dataLifetime)) {
+          throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.dataLifetime', 'a number or Lifetimes object');
+        }
+      } else if (typeof options.cacheData.dataLifetime === 'object') {
+        if (
+          options.cacheData.dataLifetime.user !== undefined &&
+          typeof options.cacheData.dataLifetime.user !== 'number'
+        ) {
+          throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.dataLifetime.user', 'a number or undefined');
+        }
+        if (
+          options.cacheData.dataLifetime.channel !== undefined &&
+          typeof options.cacheData.dataLifetime.channel !== 'number'
+        ) {
+          throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.dataLifetime.channel', 'a number or undefined');
+        }
+        if (
+          options.cacheData.dataLifetime.guild !== undefined &&
+          typeof options.cacheData.dataLifetime.guild !== 'number'
+        ) {
+          throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.dataLifetime.guild', 'a number or undefined');
+        }
+        if (
+          options.cacheData.dataLifetime.emoji !== undefined &&
+          typeof options.cacheData.dataLifetime.emoji !== 'number'
+        ) {
+          throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.dataLifetime.emoji', 'a number or undefined');
+        }
+      } else {
+        throw new TypeError('CLIENT_INVALID_OPTION', 'The cacheData.dataLifetime', 'a number or Lifetimes object');
```

Comment:
```suggestion
        throw new TypeError('CLIENT_INVALID_OPTION', 'cacheData.dataLifetime', 'a number or Lifetimes object');
```
the same for all these occourances (i can spell)

Reply:
Fixed it